packages/@aws-cdk/aws-events-targets/lib/ecs-task.ts

```diff
@@ -139,8 +138,14 @@ export class EcsTask implements events.IRuleTarget {
       return;
     }
 
+    if (!cdk.Construct.isConstruct(this.taskDefinition)) {
+      throw new Error('Cannot create a security group for ECS task. ' +
+        'The task definition in ECS task is not a Construct. ' +
+        'Please pass a taskDefinition as a Construct in EcsTaskProps.');
+    }
+
     let securityGroup = props.securityGroup || this.taskDefinition.node.tryFindChild('SecurityGroup') as ec2.ISecurityGroup;
```

Comment:
As the [last comment mentioned](https://github.com/aws/aws-cdk/pull/13293#discussion_r583465560), we'd better to see if `this.taskDefinition.node.xxx` works. Haven't found a good way to test. Open to suggestion.